components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -324,6 +269,102 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{RequeueAfter: culler.GetRequeueTime()}, nil
 }
 
+func updateNotebookStatus(r *NotebookReconciler, nb *v1beta1.Notebook,
```

Comment:
Lastly let's introduce an `updateNotebookStatus` function whose sole purpose will be to use the above `createNotebookStatus` and apply the updated status to K8s.

The signature of this function can be
```golang
func updateNotebookStatus(r *NotebookReconciler, nb *v1beta1.Notebook,
	sts *appsv1.StatefulSet, pod *corev1.Pod, req ctrl.Request) error {
```

Reply:
Done!